### PR TITLE
UCT/DC/MLX5: Add configuration for full handshake mode

### PR DESCRIPTION
**Pull request by yosefe**
# Why
Provide a method to tweak DC transport properties from UCX

### CODE REVIEW

File: src/uct/ib/dc/dc_mlx5.c [hunk 1]
```diff
@@ -1377,7 +1397,7 @@ ucs_status_t uct_dc_mlx5_iface_keepalive_init(uct_dc_mlx5_iface_t *iface)
     }
 
     dci_index = uct_dc_mlx5_iface_total_ndci(iface);
-    status    = uct_dc_mlx5_iface_create_dci(iface, 0, dci_index, 0);
+    status    = uct_dc_mlx5_iface_create_dci(iface, 0, dci_index, 0, 1);
```
Comment (hoopoepg) [inline src/uct/ib/dc/dc_mlx5.c:1400]: always use full handshake?

File: src/uct/ib/mlx5/dv/ib_mlx5_ifc.h [hunk 2]
```diff
@@ -1251,7 +1252,9 @@ struct uct_ib_mlx5_qpc_bits {
     uint8_t         counter_set_id[0x8];
     uint8_t         uar_page[0x18];
 
-    uint8_t         reserved_at_80[0x8];
+    uint8_t         reserved_at_80[0x3];
+    uint8_t         full_handshake[0x1];
```
Comment (hoopoepg) [inline src/uct/ib/mlx5/dv/ib_mlx5_ifc.h:1256]: minor: why used uint8[1] instead of just uint8 value?

**Review by hoopoepg — APPROVED** [review]

File: src/uct/ib/mlx5/dv/ib_mlx5_ifc.h (hunk 2 above)
Comment (dmitrygx) [reply src/uct/ib/mlx5/dv/ib_mlx5_ifc.h:1256]: I guess, It is aligned with PRM definition

**Review by dmitrygx — APPROVED** [review]

File: src/uct/ib/dc/dc_mlx5.h [hunk 3]
```diff
@@ -66,8 +66,14 @@ typedef enum {
...
-    UCT_DC_MLX5_IFACE_FLAG_KEEPALIVE = UCS_BIT(0), /**< keepalive dci is created */
-    UCT_DC_MLX5_IFACE_FLAG_UIDX      = UCS_BIT(1), /**< uidx is set to dci idx */
+    /** Keepalive dci is created */
+    UCT_DC_MLX5_IFACE_FLAG_KEEPALIVE                = UCS_BIT(0),
+
+    /** Enable full handshake for keepalive DCI */
+    UCT_DC_MLX5_IFACE_FLAG_KEEPALIVE_FULL_HANDSHAKE = UCS_BIT(1),
+
+    /** uidx is set to dci idx */
+    UCT_DC_MLX5_IFACE_FLAG_UIDX                     = UCS_BIT(2),
```
Comment (dmitrygx) [inline src/uct/ib/dc/dc_mlx5.h:76]: no need trailing comma

**Review by dmitrygx — APPROVED** [review]

**Review by hoopoepg — APPROVED** [review]

**Review by brminich — APPROVED** [review]

**Review by dmitrygx — APPROVED** [review]

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
bot:pipe:retes

**Comment by yosefe** [comment]
bot:pipe:retest

**Pull request merged**